pkg/collector/corechecks/system/battery/battery_windows.go

```diff
@@ -0,0 +1,422 @@
+
+	if len(info.PowerState) > 0 {
+		sender.Gauge("system.battery.power_state", 1, "", info.PowerState)
+	} else {
+		sender.Gauge("system.battery.power_state", 0, "", []string{"power_state:unknown"})
+	}
+
+	sender.Commit()
+	return nil
+}
+
+// hasBatteryAvailable checks if at least one battery device is present
+func hasBatteryAvailable() (bool, error) {
+	hdev, err := windows.SetupDiGetClassDevsEx(&GUID_DEVCLASS_BATTERY, "", 0, windows.DIGCF_PRESENT|windows.DIGCF_DEVICEINTERFACE, 0, "")
+	if err != nil {
+		return false, fmt.Errorf("SetupDiGetClassDevs: %w", err)
+	}
+	defer func() {
+		err := windows.SetupDiDestroyDeviceInfoList(hdev)
+		if err != nil {
+			log.Errorf("Error destroying device info list: %v", err)
+		}
+	}()
+
+	var ifData winutil.SP_DEVICE_INTERFACE_DATA
+	ifData.CbSize = uint32(unsafe.Sizeof(ifData))
+
+	err = winutil.SetupDiEnumDeviceInterfaces(hdev, &GUID_DEVCLASS_BATTERY, 0, &ifData)
+	if err != nil {
+		// No battery devices found
+		log.Debugf("No battery devices found")
+		return false, nil
+	}
+
+	// At least one battery device exists
+	log.Debugf("At least one battery device exists")
+	return true, nil
+}
+
+func queryBatteryInfo() (*BatteryInfo, error) {
+	info := &BatteryInfo{}
+	hdev, err := windows.SetupDiGetClassDevsEx(&GUID_DEVCLASS_BATTERY, "", 0, windows.DIGCF_PRESENT|windows.DIGCF_DEVICEINTERFACE, 0, "")
+	if err != nil {
+		return nil, fmt.Errorf("SetupDiGetClassDevs: %w", err)
+	}
+	defer func() {
+		err := windows.SetupDiDestroyDeviceInfoList(hdev)
+		if err != nil {
+			log.Errorf("Error destroying device info list: %v", err)
+		}
+	}()
+
+	var ifData winutil.SP_DEVICE_INTERFACE_DATA
+	ifData.CbSize = uint32(unsafe.Sizeof(ifData))
+
+	for i := uint32(0); ; i++ {
+		err = winutil.SetupDiEnumDeviceInterfaces(hdev, &GUID_DEVCLASS_BATTERY, i, &ifData)
+		if err != nil {
+			// No more interfaces
+			log.Debugf("No more interfaces found")
+			break
+		}
+
+		// First call: get required size
+		var required uint32
+		err = winutil.SetupDiGetDeviceInterfaceDetail(hdev, &ifData, nil, 0, &required)
+		if err != nil && err != windows.ERROR_INSUFFICIENT_BUFFER {
+			log.Errorf("Error getting device interface detail: %v", err)
+			continue
+		}
+
+		// Allocate buffer
+		buf := make([]byte, required)
+
+		// Windows SP_DEVICE_INTERFACE_DETAIL_DATA structure:
+		//   DWORD cbSize (4 bytes)
+		//   WCHAR DevicePath[1] (2 bytes for first char)
+		// The structure size is: sizeof(uint32) + sizeof(uint16) = 6 bytes
+		// Windows aligns this to pointer size on 64-bit (8 bytes), but structure is still 6 bytes on 32-bit
+		// ref: https://stackoverflow.com/questions/10728644
+		cbSizeBase := unsafe.Sizeof(uint32(0)) + unsafe.Sizeof(uint16(0))
+		ptrSize := unsafe.Sizeof(uintptr(0))
+		cbSizeValue := uint32(cbSizeBase)
+		if ptrSize > cbSizeBase {
+			cbSizeValue = uint32(ptrSize)
+		}
+
+		// Write cbSize directly to the buffer (first 4 bytes)
+		*(*uint32)(unsafe.Pointer(&buf[0])) = cbSizeValue
```

Comment:
I am missing something here, I thought that it should be required value
`*(*uint32)(unsafe.Pointer(&buf[0])) = required`